standard/patterns.md

```diff
@@ -40,11 +40,13 @@ single_variable_designation
     ;
 ```
 
-The runtime type of the value is tested against the *type* in the pattern. If it is of that runtime type (or some subtype), the pattern *matches* that value. This pattern form never matches a `null` value.
+The runtime type of the value is tested against the *type* in the pattern using the same rules specified in the is-type operator (§12.12.12.1) returns true. If the test succeeds succeeds, the pattern *matches* that value. It is a compile-time error if the *type* is a nullable value type (§8.3.12). This pattern form never matches a `null` value.
+
+> *Note*: The is-type expression `e is T` and the declaration pattern `e is T _` are the same when `T` isn't a nullable type. *end note*
```

Comment:
"the same" to "equivalent" perhaps?

"isn't a nullable type" sounds a bit strange to me too - would "isn't a nullable value type" be better? (Aside from anything else, when we have nullable reference types, presumably it's *still* the same for reference types, whether the compiler thinks it's nullable or not.)

Reply:
I updated to use the "equivalent" language.

IF I specify "nullable value type", we'll need to update for nullable reference types in C# 8.0. `e is string?` isn't valid, only `e is string`.